contrib/create-waku/cli.js

```diff
@@ -1,46 +1,148 @@
 #!/usr/bin/env node
 
-const path = require("node:path");
-const fs = require("node:fs");
-const https = require("node:https");
+import fs from 'node:fs'
+import path from 'node:path'
+import prompts from 'prompts'
+import { red, green, bold } from 'kolorist'
 
-const dirName = "waku-example";
+import emptyDir from './emptyDir.js'
+import renderTemplate from './renderTemplate.js'
 
-if (fs.existsSync(dirName)) {
-  console.error(`Directory "${dirName}" already exists!`);
-  process.exit(1);
+function isValidPackageName(projectName) {
+  return /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/.test(
+    projectName
+  )
 }
 
-const baseUrl =
-  "https://raw.githubusercontent.com/dai-shi/waku/v0.14.0/examples/01_counter/";
-
-const files = `
-package.json
-tsconfig.json
-src/main.tsx
-src/entries.ts
-src/index.html
-src/components/App.tsx
-src/components/Counter.tsx
-`
-  .split(/\s/)
-  .filter((file) => file);
-
-const getFiles = (index = 0) => {
-  const file = files[index];
-  if (!file) return;
-  const destFile = path.join(dirName, file.replace("/", path.sep));
-  fs.mkdirSync(path.dirname(destFile), { recursive: true });
-  https.get(baseUrl + file, (res) => {
-    res.pipe(fs.createWriteStream(destFile));
-    res.on("end", () => getFiles(index + 1));
-  });
-};
-
-getFiles();
-
-process.on("exit", (code) => {
-  if (!code) {
-    console.info(`Done! Change directory "${dirName}"`);
+function toValidPackageName(projectName) {
+  return projectName
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, '-')
+    .replace(/^[._]/, '')
+    .replace(/[^a-z0-9-~]+/g, '-')
+}
+
+// if the dir is empty or not exist
+function canSafelyOverwrite(dir) {
+  return !fs.existsSync(dir) || fs.readdirSync(dir).length === 0
+}
+
+async function init() {
+  const cwd = process.cwd();
+
+  let targetDir;
+  let defaultProjectName = 'waku-project'
+
+  const CHOICES = fs.readdirSync(path.resolve(cwd, '../../examples'))
```

Comment:
> This doesn't work as we don't publish examples dir.
> 
> We have two options.
> 
> 1. add build step to copy examples dir
> 2. download from github
> 
> Eventually, we should go with 1 because it fixes #81, but for now 2 is also fine. If we were to go with 1, we would like to make the code in TypeScript and transpile with SWC in the build step.

Summary,
- If we ``npm run build:esbuild`` an ``out.js`` file generated.
- Running ``npm run build:estemplate`` generates an ``copy.js`` file.
- If we run ``node copy.js`` templates copied from ``example`` dir and a ``template`` folder generated.
- Finally, running ``node cli.js`` works fine.

Problem with transpiling with ``swr``,
- If we transpiling files with ``swr`` then the ``imports`` in output files are not working ex,
```js
import { renderTemplate } from './renderTemplate'
```
This import statement would not be able to resolve the file.

Is, I'm missing something about this ``swr``.

Reply:
Okay, I don't mind using esbuild for now. I just thought SWC might work better and we also use it in `waku`. We can try that later.

Can you move TS files into `./src` folder please?

Reply:
Yeah, sure we can look around it in the near future.